broker/src/main/java/io/camunda/zeebe/broker/system/partitions/PartitionAdminControlImpl.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.1. You may not use this file
+ * except in compliance with the Zeebe Community License 1.1.
+ */
+package io.camunda.zeebe.broker.system.partitions;
+
+import io.camunda.zeebe.broker.exporter.stream.ExporterDirector;
+import io.camunda.zeebe.broker.system.partitions.impl.AsyncSnapshotDirector;
+import io.camunda.zeebe.broker.system.partitions.impl.PartitionProcessingState;
+import io.camunda.zeebe.engine.processing.streamprocessor.StreamProcessor;
+import java.io.IOException;
+import java.util.function.Supplier;
+
+public class PartitionAdminControlImpl implements PartitionAdminControl {
+
+  private final Supplier<StreamProcessor> streamProcessorSupplier;
+  private final Supplier<ExporterDirector> exporterDirectorSupplier;
+  private final Supplier<AsyncSnapshotDirector> snapshotDirectorSupplier;
+  private final Supplier<PartitionProcessingState> partitionProcessingStateSupplier;
+
+  public PartitionAdminControlImpl(
```

Comment:
I want to get to a point where a `PartitionContext` is immutable, but can be swapped (in `ZeebePartition`) at runtime, e.g. after a transition. 

The `PartitionAdminControlImpl` shall only be created once. That's why it gets lookup methods instead of instances for its dependencies. Whether the lookup is for four individual things or one special interface is secondary. I wouldn't want to use `PartitionContext` because at least now it is both two big and missing some of the stuff we need.